### PR TITLE
Tiaf bucket top level fix

### PR DESCRIPTION
**Pull request by jonawals**
Currently, it is possible for different repos using the same s3 bucket for persistent storage to trample over each other's historic data. This fixes this issue by adding an optional top level directory script parameter to use as the top level directory for branches in the bucket.

Additionally, an optional build number script parameter has been added to the MARS job document to make cross referencing TIAF documents in MARS with specific builds.

### CODE REVIEW

File: scripts/build/Jenkins/Jenkinsfile [hunk 1]
```diff
@@ -247,7 +247,7 @@ def CheckoutRepo(boolean disableSubmodules = false) {
                 [$class: 'CheckoutOption', timeout: 60]
             ],
             userRemoteConfigs: scm.userRemoteConfigs
-       ]
+       ]
```
Comment (amznestebanpapp) [inline scripts/build/Jenkins/Jenkinsfile:250]: cosmetic: seems this is adding unnecessary spaces

File: scripts/build/Jenkins/Jenkinsfile [hunk 2]
```diff
@@ -594,6 +594,7 @@ try {
                     // repositoryName is the full repository name
                     repositoryName = (repositoryUrl =~ /https:\/\/github.com\/(.*)\.git/)[0][1]
                     env.REPOSITORY_NAME = repositoryName
+                    env.REPOSITORY_URL = repositoryUrl
```
Comment (amznestebanpapp) [inline scripts/build/Jenkins/Jenkinsfile:597]: `REPOSITORY_NAME` may be a better one for what you want

**Review by amznestebanpapp — CHANGES_REQUESTED** [review]

File: scripts/build/Jenkins/Jenkinsfile (hunk 1 above)
Comment (jonawals) [reply scripts/build/Jenkins/Jenkinsfile]: fixed 👍

**Review by amznestebanpapp — APPROVED** [review]

File: scripts/build/Jenkins/Jenkinsfile (hunk 2 above)
Comment (jonawals) [reply scripts/build/Jenkins/Jenkinsfile]: great suggestion! fixed 👍

**Pull request merged**